src/encoded/static/components/search.js

```diff
@@ -783,34 +804,72 @@ class Facet extends React.Component {
+                                </div>
                             </div>
                         : null}
-                        {(terms.length > 5 && !moreTermSelected) ?
-                            <div className="pull-right">
-                                {/* Display the "+ See more" button if more than five terms exist for this facet */}
-                                <small>
-                                    <button type="button" className={seeMoreClass} data-toggle="collapse" data-target={`#${termID}`} onClick={this.handleClick} />
-                                </small>
+                        {/* If user has searched using the typeahead, we will not display the full set of facet terms, just those matching the search */}
+                        {(this.state.filteredTerms !== null) ?
+                            <div>
+                                {/* Display error message if there is a search but no results found */}
+                                {(this.state.filteredTerms.length === 0) ?
+                                    <div className="searcherror">
+                                        Try a different search term for results.
+                                    </div>
+                                :
+                                    <div className="terms-block">
+                                        {/* List of results does not overflow top on initialization */}
+                                        <div className="top-shading hide-shading" />
+                                        {/* List of filtered terms */}
+                                        <div className={`term-list search${titleComponent.replace(/\s+/g, '')}`} onScroll={shadeOverflowOnScroll}>
+                                            {this.state.filteredTerms.map(term =>
+                                                <TermComponent {...this.props} key={term.key} term={term} filters={filters} total={total} canDeselect={canDeselect} statusFacet={statusFacet} />
+                                            )}
+                                        </div>
+                                        {/* Only show bottom shading when list of results overflows */}
+                                        <div className={`shading ${(this.state.filteredTerms.length < displayedTermsCount) ? 'hide-shading' : ''}`} />
+                                    </div>
+                                }
                             </div>
-                        : null}
+                        :
+                            <div>
+                                {/* If the user has not searched, we will display the full set of facet terms */}
+                                {((terms.length && terms.some(term => term.doc_count)) || (field.charAt(field.length - 1) === '!')) ?
```

Comment:
Our AirBnB config requires `terms.length` to be compared against an integer instead of truthyness. https://github.com/airbnb/javascript#comparison--shortcuts